cppapi/server/device.cpp

```diff
@@ -3229,7 +3229,43 @@ void DeviceImpl::add_attribute(Tango::Attr *new_attr)
 
 	if (new_attr->is_fwd() == true)
 	{
-		dev_attr->add_fwd_attribute(device_name,device_class,i,new_attr);
+		// If forwarded attribute is dynamically created and was constructed without specifying
```

Comment:
This code should be extracted into method `add_forwarded_attribute(Tango::Attr* fwd_attr)`